source/common/common/utility.h

```diff
@@ -578,7 +579,11 @@ template <class Value> struct TrieLookupTable {
       current = current->entries_[c].get();
       key++;
     }
+    if (current->value_) {
```

Comment:
This might change the behavior for current callers. Should `overwrite_existing` instead be a param?